app/models/manageiq/providers/azure/cloud_manager/provision/cloning.rb

```diff
@@ -82,13 +80,22 @@ def prepare_for_clone_task
             :caching      => 'ReadWrite',
             :osType       => os
           }
-        },
-        :networkProfile  => {
-          :networkInterfaces => [{:id => nic_id}],
         }
       }
     }
 
+    # The -1 value is set in ProvisionWorkflow to distinguish between the
+    # desire for a new Public IP address vs a private IP.
+    #
+    if floating_ip
+      nic_id = associated_nic || create_nic(true)
+    else
+      public_ip = options[:floating_ip_address].first == -1
+      nic_id = create_nic(public_ip)
```

Comment:
so the `else` statement will be executed if floating_ip is `nil` which would be either:
- the floating_ip dropdown was ignored altogether 
or
- "New" was selected and therefore the look up in the options helper returns `nil` - refer to lookup [here](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/cloud_manager/provision/options_helper.rb#L19)

Reply:
Correct.